presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -960,6 +960,13 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   std::vector<std::string> aggregateNames;
   std::vector<core::AggregationNode::Aggregate> aggregates;
 
+  // Do not allow streaming aggregations
+  auto streamable = !node->preGroupedVariables.empty()
+      && node->groupingSets.groupingSetCount == 1
+      && node->groupingSets.globalGroupingSets.empty();
+  VELOX_CHECK(!streamable,
```

Comment:
Is it possible to just use a regular aggregation instead of failing?

Reply:
@rschlussel : Yes. You are correct. 
There is a nuance though, HashAggregation also has preGrouping related optimizations which streamable should disable.

That is the fix in https://github.com/prestodb/presto/pull/22998.